docs/source/monitoring.md

```diff
@@ -1,76 +1,142 @@
+      scylla/cluster: replace-with-your-scyllacluster-name
+  components:
+    prometheus:
+      storage:
+        volumeClaimTemplate:
+          spec:
+            resources:
+              requests:
+                storage: 1Gi
+    grafana:
+      exposeOptions:
+        webInterface:
+          ingress:
+            ingressClassName: haproxy
+            dnsDomains:
+            - test-grafana.test.svc.cluster.local
+            annotations:
+              haproxy-ingress.github.io/ssl-passthrough: "true"
+```
+
+For details, refer to the below command:
+```console
+$ kubectl explain scylladbmonitorings.scylla.scylladb.com/v1alpha1
+```
+
+## Deploy managed monitoring
+
+**Note**: as of v1.8, ScyllaDBMonitoring is experimental. The API is currently in version v1alpha1 and may change in future versions.
+
+### Requirements
+
+Before you can set up your ScyllaDB monitoring, you need Scylla Operator already installed in your Kubernetes cluster.
+For more information on how to deploy Scylla Operator, see:
+* [Deploying Scylla on a Kubernetes Cluster](generic.md)
+* [Deploying Scylla stack using Helm Charts](helm.md)
+
+The above example of the monitoring setup also makes use of HAProxy Ingress and Prometheus Operator.
+You can deploy them in your Kubernetes cluster using the provided third party examples. If you already have them deployed
+in your cluster, you can skip the below steps.
+
+#### Deploy Prometheus Operator
+Deploy Prometheus Operator using kubectl:
+```console
+$ kubectl -n prometheus-operator apply --server-side -f ./examples/third-party/prometheus-operator
+```
+
+##### Verify Prometheus Operator status
+```console
+$ kubectl -n prometheus-operator rollout status --timeout=5m deployments.apps/prometheus-operator
+deployment "prometheus-operator" successfully rolled out
+```
+
+#### Deploy HAProxy Ingress
+Deploy HAProxy Ingress using kubectl:
+```console
+$ kubectl -n haproxy-ingress apply --server-side -f ./examples/third-party/haproxy-ingress
+```
+
+##### Verify HAProxy Ingress status
+```console
+$ kubectl -n haproxy-ingress rollout status --timeout=5m deployments.apps/haproxy-ingress
+deployment "haproxy-ingress" successfully rolled out
+```
+
+Before you begin, update the `endpointsSelector` in `examples/monitoring/v1alpha1/scylladbmonitoring.yaml` with a label
+matching your ScyllaCluster instance name.
+
+### Deploy ScyllaDBMonitoring
+
+Deploy the monitoring setup using kubectl:
+```console
+$ kubectl apply --server-side -f ./examples/monitoring/v1alpha1/scylladbmonitoring.yaml
+```
+
+Scylla Operator will notice the new ScyllaDBMonitoring object, and it will reconcile all necessary resources.
+
+#### Verify ScyllaDBMonitoring status
+```console
+$ kubectl wait --for='condition=Available=True' scylladbmonitorings.scylla.scylladb.com/example
+scylladbmonitoring.scylla.scylladb.com/example condition met
+
+$ kubectl wait --for='condition=Progressing=False' scylladbmonitorings.scylla.scylladb.com/example
```

Comment:
with this raciness, you should check progressing first and available last. it can be still available for a while when it starts progressing and when it rolls out you won't check it again

Reply:
good point, updated